### PR TITLE
HIVE-25710: Config used to enable non-blocking TRUNCATE is not proper…

### PR DESCRIPTION
**Pull request by deniskuzZ**
…ly propagated

### CODE REVIEW

File: common/src/java/org/apache/hadoop/hive/conf/HiveConf.java [hunk 1]
```diff
@@ -3060,8 +3060,15 @@ private static void populateLlapDaemonVarsSet(Set<String> llapDaemonVarsSetLocal
         "Creates an _orc_acid_version file along with acid files, to store the version data"),
 
     HIVE_TXN_READONLY_ENABLED("hive.txn.readonly.enabled", false,
-      "Enables read-only transaction classification and related optimizations"),
-
+        "Enables read-only transaction classification and related optimizations"),
+
+    HIVE_ACID_LOCKLESS_READS_ENABLED("hive.acid.lockless.reads.enabled", false,
```
Comment (pvary) [inline common/src/java/org/apache/hadoop/hive/conf/HiveConf.java:3065]: Where will we document the requirements when this lockless reads can work?
Like `HIVE_ACID_TRUNCATE_USE_BASE` should be true

File: ql/src/java/org/apache/hadoop/hive/ql/ddl/table/misc/truncate/TruncateTableAnalyzer.java [hunk 2]
```diff
@@ -116,10 +116,13 @@ private void checkTruncateEligibility(ASTNode ast, ASTNode root, String tableNam
 
   private void addTruncateTableOutputs(ASTNode root, Table table, Map<String, String> partitionSpec)
       throws SemanticException {
-    boolean truncateKeepsDataFiles = AcidUtils.isTransactionalTable(table) &&
-        MetastoreConf.getBoolVar(conf, MetastoreConf.ConfVars.TRUNCATE_ACID_USE_BASE);
+    boolean truncateUseBase = (HiveConf.getBoolVar(conf, HiveConf.ConfVars.HIVE_ACID_TRUNCATE_USE_BASE)
+        || HiveConf.getBoolVar(conf, HiveConf.ConfVars.HIVE_ACID_LOCKLESS_READS_ENABLED))
```
Comment (pvary) [inline ql/src/java/org/apache/hadoop/hive/ql/ddl/table/misc/truncate/TruncateTableAnalyzer.java:120]: What happened with the `TRUNCATE_ACID_USE_BASE` config?
Do we need to deprecate it? Was this used for the same?

File: standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java [hunk 3]
```diff
@@ -1841,39 +1841,47 @@ public void dropTable(String catName, String dbname, String name, boolean delete
...
   private void truncateTableInternal(String catName, String dbName, String tableName,
-      List<String> partNames, String validWriteIds, long writeId)
-          throws MetaException, TException {
+      List<String> partNames, String validWriteIds, long writeId, boolean deleteData)
+          throws TException {
     Table table = getTable(catName, dbName, tableName);
     HiveMetaHook hook = getHook(table);
-    EnvironmentContext envContext = new EnvironmentContext();
+    EnvironmentContext context = new EnvironmentContext();
+    context.putToProperties("truncateSkipDataDeletion", Boolean.toString(!deleteData));
```
Comment (pvary) [inline standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java:1875]: Should `truncateSkipDataDeletion` be a constant?

File: standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/conf/MetastoreConf.java [hunk 4]
```diff
@@ -1397,9 +1397,6 @@ public static ConfVars getMetaConf(String name) {
             " If org.apache.hive.hcatalog.listener.DbNotificationListener is configured along with other transactional event" +
             " listener implementation classes, make sure org.apache.hive.hcatalog.listener.DbNotificationListener is placed at" +
             " the end of the list."),
-    TRUNCATE_ACID_USE_BASE("metastore.acid.truncate.usebase", "hive.metastore.acid.truncate.usebase", false,
```
Comment (pvary) [inline standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/conf/MetastoreConf.java:1400]: How old is this config? Is this released somewhere? Shall we deprecate it, or it is ok to remove it?

File: common/src/java/org/apache/hadoop/hive/conf/HiveConf.java (hunk 1 above)
Comment (deniskuzZ) [reply common/src/java/org/apache/hadoop/hive/conf/HiveConf.java:3065]: HIVE_ACID_LOCKLESS_READS_ENABLED is independent from HIVE_ACID_TRUNCATE_USE_BASE. HIVE_ACID_LOCKLESS_READS_ENABLED enables group of non-blocking operations at once like drop/truncate table/partition.

File: ql/src/java/org/apache/hadoop/hive/ql/ddl/table/misc/truncate/TruncateTableAnalyzer.java (hunk 2 above)
Comment (deniskuzZ) [reply ql/src/java/org/apache/hadoop/hive/ql/ddl/table/misc/truncate/TruncateTableAnalyzer.java:120]: It didn't work (only in tests). Not sure if we need to deprecate old property.

File: standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java (hunk 3 above)
Comment (deniskuzZ) [reply standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStoreClient.java:1875]: yea, I'll fix that

File: standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/conf/MetastoreConf.java (hunk 4 above)
Comment (deniskuzZ) [reply standalone-metastore/metastore-common/src/main/java/org/apache/hadoop/hive/metastore/conf/MetastoreConf.java:1400]: it's disabled downstream + it doesn't work correctly as the property activating it is not propagated.

File: ql/src/java/org/apache/hadoop/hive/ql/ddl/table/misc/truncate/TruncateTableAnalyzer.java (hunk 2 above)
Comment (pvary) [reply ql/src/java/org/apache/hadoop/hive/ql/ddl/table/misc/truncate/TruncateTableAnalyzer.java:120]: If it was never working, we do not need to deprecate it

**Review by pvary — APPROVED** [review]
I am ok with the change.
@klcopp: Do you want to review too?

**Review by klcopp — APPROVED** [review]

**Pull request merged**